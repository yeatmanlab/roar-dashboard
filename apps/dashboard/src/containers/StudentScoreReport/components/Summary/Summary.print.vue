<template>
  <div class="pt-2 mt-2 text-sm border border-gray-800 border-top-2">
    <h2 class="text-xl font-bold">Assessment Overview</h2>
    <p>{{ $t('scoreReports.roarSummary') }}</p>

    <i18n-t keypath="scoreReports.completedTasks" tag="p">
      <template #firstName>
        {{ studentFirstName }}
      </template>
    </i18n-t>
    <ul class="p-0 pl-3 md:columns-2">
      <li v-for="task in tasks" :key="task" class="pb-1">
        <strong>{{ task }}</strong>
      </li>
    </ul>
    <i18n-t keypath="scoreReports.summary" tag="p">
      <template #firstName>
        {{ studentFirstName }}
      </template>
    </i18n-t>
  </div>
</template>

<script setup>
defineProps({
  studentFirstName: {
    type: String,
    required: true,
  },
  expanded: {
    type: Boolean,
    default: false,
  },
  exportLoading: {
    type: Boolean,
    default: false,
  },
  tasks: {
    type: Array,
    required: true,
  },
});

defineEmits(['toggleExpand', 'exportPdf']);
</script>
