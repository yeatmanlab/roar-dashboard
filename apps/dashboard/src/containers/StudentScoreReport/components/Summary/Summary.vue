<template>
  <section class="pt-4 lg:pr-8">
    <p>{{ $t('scoreReports.roarSummary') }}</p>

    <i18n-t keypath="scoreReports.completedTasks" tag="p" class="mt-2">
      <template #firstName>
        {{ studentFirstName }}
      </template>
    </i18n-t>

    <ul class="p-0 list-none lg:columns-2">
      <li v-for="task in tasks" :key="task" class="pb-1">
        <strong>{{ task }}</strong>
      </li>
    </ul>

    <i18n-t keypath="scoreReports.summary" tag="p">
      <template #firstName>
        {{ studentFirstName }}
      </template>
    </i18n-t>
  </section>
</template>

<script setup>
defineProps({
  studentFirstName: {
    type: String,
    required: true,
  },
  tasks: {
    type: String,
    required: true,
  },
});
</script>

<style scoped>
/* @TODO: Migrate to tailwind classes once yeatmanlab/roar-project-management/issues/321 is resolved */
ul li {
  position: relative;
}
ul li::before {
  content: '\e909'; /* Unicode for pi-check */
  font-family: 'PrimeIcons';
  font-size: 0.75rem;
  color: var(--gray-500);
  margin-right: 0.5rem;
}
</style>
