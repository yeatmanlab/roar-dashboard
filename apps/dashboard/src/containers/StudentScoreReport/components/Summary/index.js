export { default as SummaryScreen } from './Summary.vue';
export { default as SummaryPrint } from './Summary.print.vue';
