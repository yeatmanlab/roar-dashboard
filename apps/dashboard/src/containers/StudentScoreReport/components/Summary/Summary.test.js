import { describe, it, expect, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import Summary from './Summary.vue';

describe('Summary.vue', () => {
  let i18n;

  beforeEach(() => {
    i18n = createI18n({
      legacy: false,
      locale: 'en',
      messages: {
        en: {
          scoreReports: {
            roarSummary: 'This is a summary of assessment results.',
            completedTasks: '{firstName} has completed the following assessments:',
            summary: 'Based on these results, {firstName} shows strong performance.',
          },
        },
      },
    });
  });

  describe('snapshots', () => {
    it('renders with single task', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'John',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with multiple tasks', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'Sarah',
          tasks: 'Single Word Reading, Sentence Reading, Phonological Awareness',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders with different student name', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'Michael',
          tasks: 'Single Word Reading, Phonics',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('component behavior', () => {
    it('displays student first name', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'John',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('John');
    });

    it('displays tasks list', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'John',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('Single Word Reading');
    });

    it('renders list with proper structure', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'John',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      const list = wrapper.find('ul');
      expect(list.exists()).toBe(true);
      expect(list.classes()).toContain('list-none');
    });

    it('renders summary paragraph', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'John',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('Based on these results, John shows strong performance');
    });

    it('displays completed tasks message', () => {
      const wrapper = mount(Summary, {
        props: {
          studentFirstName: 'Sarah',
          tasks: 'Single Word Reading',
        },
        global: {
          plugins: [i18n],
        },
      });

      expect(wrapper.text()).toContain('Sarah has completed the following assessments');
    });
  });
});
