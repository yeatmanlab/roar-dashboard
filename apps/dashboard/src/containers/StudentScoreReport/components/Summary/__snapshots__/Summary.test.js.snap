// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Summary.vue > snapshots > renders with different student name 1`] = `
"<section data-v-7eb30534="" class="pt-4 lg:pr-8">
  <p data-v-7eb30534=""></p>
  <p data-v-7eb30534="" class="mt-2">Michael has completed the following assessments:</p>
  <ul data-v-7eb30534="" class="p-0 list-none lg:columns-2">
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">S</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">l</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">W</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">r</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">R</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">,</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">P</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">h</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">c</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">s</strong></li>
  </ul>
  <p data-v-7eb30534="">Based on these results, Michael shows strong performance.</p>
</section>"
`;

exports[`Summary.vue > snapshots > renders with multiple tasks 1`] = `
"<section data-v-7eb30534="" class="pt-4 lg:pr-8">
  <p data-v-7eb30534=""></p>
  <p data-v-7eb30534="" class="mt-2">Sarah has completed the following assessments:</p>
  <ul data-v-7eb30534="" class="p-0 list-none lg:columns-2">
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">S</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">l</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">W</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">r</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">R</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">,</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">S</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">t</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">c</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">R</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">,</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">P</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">h</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">l</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">c</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">l</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">A</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">w</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">r</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">s</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">s</strong></li>
  </ul>
  <p data-v-7eb30534="">Based on these results, Sarah shows strong performance.</p>
</section>"
`;

exports[`Summary.vue > snapshots > renders with single task 1`] = `
"<section data-v-7eb30534="" class="pt-4 lg:pr-8">
  <p data-v-7eb30534=""></p>
  <p data-v-7eb30534="" class="mt-2">John has completed the following assessments:</p>
  <ul data-v-7eb30534="" class="p-0 list-none lg:columns-2">
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">S</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">l</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">W</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">o</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">r</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534=""> </strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">R</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">e</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">a</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">d</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">i</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">n</strong></li>
    <li data-v-7eb30534="" class="pb-1"><strong data-v-7eb30534="">g</strong></li>
  </ul>
  <p data-v-7eb30534="">Based on these results, John shows strong performance.</p>
</section>"
`;
